src/main/kotlin/org/opensearch/replication/task/shard/ShardReplicationTask.kt

```diff
@@ -158,6 +158,7 @@ class ShardReplicationTask(id: Long, type: String, action: String, description:
         * it continues to be called even after the task is completed.
          */
         clusterService.removeListener(clusterStateListenerForTaskInterruption)
+        this.followerClusterStats.stats.remove(followerShardId)
```

Comment:
This is a tricky block. When a shardTask is closed and opened on a new node during B/G, how do we propagate follower cluster stats?

Reply:
Its fine if we cant address this by tomorrow and come to common understanding that they will be reset on B/G

Reply:
yes, this is the understanding bhanu. We want to discard the metrics from older nodes.